### ISSUE TITLE
Title modal - accessibility issue

### ISSUE THREAD

**Issue by abdermaiza** [twbs/bootstrap#37179]
### Prerequisites

- [X] I have [searched](https://github.com/twbs/bootstrap/issues?utf8=%E2%9C%93&q=is%3Aissue) for duplicate or closed issues
- [X] I have [validated](https://html5.validator.nu/) any HTML to avoid common problems
- [X] I have read the [contributing guidelines](https://github.com/twbs/bootstrap/blob/main/.github/CONTRIBUTING.md)

### Describe the issue

The modal title is now a H5, it should be a H1 or an element with role="heading" aria-level="1".

### Reduced test cases

https://getbootstrap.com/docs/5.0/components/modal/

### What operating system(s) are you seeing the problem on?

Windows

### What browser(s) are you seeing the problem on?

Chrome

### What version of Bootstrap are you using?

5.0

**Comment by mdo** [comment]
There's no reason it needs to be a specific heading, unless you're trying to align the modal to it's place in a heading hierarchy, in which case you should use whatever you want :).

**Comment by abdermaiza** [comment]
I though because opening a modal is creating a new context (like a new page), it's more coherent to have a H1 here...

**Comment by patrickhlauke** [comment]
I'll have a look at this. as modal dialogs create their own little world/document/structure, it would make sense that the title is the primary heading in this new "island" of content. particularly if the underlying page is hidden from AT (which, I think ... though need to check) we do in BS.

**Comment by patrickhlauke** [comment]
to be clear though. not necessarily a failure of WCAG or anything currently. more of a best practice exploration

**Comment by martinbean** [comment]
To be honest, I’ve always changed the mark-up to be `<h1 class="h5 modal-title">` when copying the example from the docs, as it makes more sense that semantically the heading is the top-level heading in the context of the modal.

**Comment by leecollings-sb** [comment]
This should never have been changed to a H1, as the modal code is on the page along with the actual H1. There should NEVER be more than one H1 ont he page, this will massively negatively impact the users page ranking for this page.

This needs addressing ASAP.

**Comment by abdermaiza** [comment]
we are targeting more assistive technologies, because when you open a modal, the rest of the page is hidden with aria-hidden set to true, so for them, there will be no other H1 on the page....
But if you really cannot have it, then jus use a <p role="heading" aria-level="1"> 

:)

**Comment by leecollings-sb** [comment]
It's not hidden for any page crawlers that will see multiple H1s and then flag this up as a big issue.

The simple solution is to keep the titles of modals to H2 or lower (which you are allowed to have multiple on the page).

Don't use H1 because it's not the title of the page!

**Comment by patrickhlauke** [comment]
and again https://www.google.com/search?q=google+ranking+multiple+h1

**Comment by leecollings-sb** [comment]
You're arguing with me about a change that shouldn't have happened. You stated it was a bug fix, but it hasn't fixed anything because nothing was wrong with how it was?

Yet again, another example of how BS devs just don't care what anyone else thinks and do what they want anyway.

I've said what I wanted to say, my point is valid, andthe new docs are unhelpful to those who will be use them. Mark my words.

**Comment by patrickhlauke** [comment]
> You stated it was a bug fix, but it hasn't fixed anything because nothing was wrong with how it was?

it's a semantic/structural bug, and it affects users of assistive technologies.
